### PR TITLE
Add CQL role handling

### LINKED ISSUE #22: Provide options for provisioning the default user through the service broker
Need to have the service broker provide some kind of option to grant a certain level of permissions to the user that gets created.  With RBAC now enabled, the default user that gets created by the Service Broker has no permissions and a manual step must be executed as the cassandra admin user to grant permissions after the service key has been created.

### PR DESCRIPTION
**Pull request by ramkumarvs**
Tested by making the service broker point to k8s yugaware

Created Service
```
curl -X PUT \
  http://localhost:8080/v2/service_instances/270966a5-974e-4d46-a3bb-c3d480e5011m \
  -H 'Content-Type: application/json' \
  -d '{
	"service_id": "demo",
	"plan_id": "xsmall",
	"parameters": {
		"universe_name": "demo",
		"provider_type": "kubernetes",
		"kube_provider": "gke"
	}
}'
```
Create service binding with additional params 
```
curl -X PUT \
  http://localhost:8080/v2/service_instances/270966a5-974e-4d46-a3bb-c3d480e5011m/service_bindings/bind8.3 \
  -H 'Content-Type: application/json' \
  -H 'X-AUTH-TOKEN: 6d502312-fcc0-4561-b194-a8e14fd60543' \
  -H 'cache-control: no-cache' \
  -d '{
	"binding_id": "bind-4",
	"service_id": "270966a5-974e-4d46-a3bb-c3d480e5011m",
	"plan_id": "xsmall",
	"parameters": {
		"role": "read_only"
		
	}
}'
```

Fixes #22

### CODE REVIEW

File: src/main/java/com/yugabyte/servicebroker/utils/YCQLClient.java [hunk 1]
```diff
@@ -56,6 +75,20 @@ public YCQLClient(List<HostAndPort> serviceHosts,
                             adminCreds.get("password"));
     Cluster cluster = builder.build();
     session = cluster.connect();
+    createSystemRoles();
+    return session;
+  }
+
+  private void createSystemRoles() {
+    // We will also create two roles in the system, one is admin and other is
+    // readonly, off course users can add their own roles and grant them.
```
Comment (bbaddepudi) [inline src/main/java/com/yugabyte/servicebroker/utils/YCQLClient.java]: off -> of

File: src/main/java/com/yugabyte/servicebroker/utils/YBClient.java [hunk 2]
```diff
@@ -26,13 +26,24 @@
...
+      }
+      return null;
+    }
   }
 
   private List<HostAndPort> serviceHostPorts;
   protected List<HostAndPort> getServiceHostPorts() { return serviceHostPorts; }
   protected abstract int getDefaultPort();
-  protected abstract Map<String, String> createAuth();
+  protected abstract Map<String, String> createAuth(Map<String, Object> parameters);
```
Comment (bbaddepudi) [inline src/main/java/com/yugabyte/servicebroker/utils/YBClient.java:49]: Could add a comment on what `parameters` are expected...

File: src/main/java/com/yugabyte/servicebroker/utils/YCQLClient.java [hunk 3]
```diff
@@ -64,14 +97,19 @@ protected int getDefaultPort() {
...
+    session = getSession();
     // Cassandra username are all lowercase and avoid numbers in the beginning.
     String username = generateRandomString(true).toLowerCase();
     String password = generateRandomString(false);
     String createRole = "CREATE ROLE " + username + " with superuser = false " +
         "and login = true and password = '" + password + "'";
+
     session.execute(createRole);
-    session.close();
+    String role = (String) parameters.getOrDefault("role", "admin");
```
Comment (bbaddepudi) [inline src/main/java/com/yugabyte/servicebroker/utils/YCQLClient.java:109]: Do we need to check sanity of this `role` if it is not the default admin? As in that role exists etc..

**Review by bbaddepudi — COMMENTED** [review]
Few ques/comments.

File: src/main/java/com/yugabyte/servicebroker/utils/YCQLClient.java (hunk 3 above)
Comment (ramkumarvs) [reply src/main/java/com/yugabyte/servicebroker/utils/YCQLClient.java:109]: @bbaddepudi `createSystemRoles()` will make sure to create two preseeded roles (Admin and Read_Only)

**Review by ajcaldera1 — APPROVED** [review]

**Pull request merged**